pkg/cmd/build/build.go

```diff
@@ -0,0 +1,443 @@
+			named, err := dockerreference.ParseNormalizedNamed(tag)
+			if err != nil {
+				return "", nil, false, "", nil, nil, err
+			}
+			tags[idx] = dockerreference.TagNameOnly(named).String()
+		}
+	} else if len(tags) == 0 {
+		output = output + ",dangling-name-prefix=<none>"
+	}
+
+	buildctlArgs = buildkitutil.BuildctlBaseArgs(buildkitHost)
+
+	progressValue := options.Progress
+
+	buildctlArgs = append(buildctlArgs, []string{
+		"build",
+		"--progress=" + progressValue,
+		"--frontend=dockerfile.v0",
+		"--local=context=" + buildContext,
+		"--output=" + output,
+	}...)
+
+	filename := options.File
+
+	dir := buildContext
+	file := buildkitutil.DefaultDockerfileName
+	if filename != "" {
+		if filename == "-" {
+			var err error
+			dir, err = buildkitutil.WriteTempDockerfile(options.StdIn)
+			if err != nil {
+				return "", nil, false, "", nil, nil, err
+			}
+			cleanup = func() {
+				os.RemoveAll(dir)
+			}
+		} else {
+			dir, file = filepath.Split(filename)
+		}
+
+		if dir == "" {
+			dir = "."
+		}
+	}
+	dir, file, err = buildkitutil.BuildKitFile(dir, file)
+	if err != nil {
+		return "", nil, false, "", nil, nil, err
+	}
+
+	buildctlArgs = append(buildctlArgs, "--local=dockerfile="+dir)
+	buildctlArgs = append(buildctlArgs, "--opt=filename="+file)
+
+	target := options.Target
+	if target != "" {
+		buildctlArgs = append(buildctlArgs, "--opt=target="+target)
+	}
+
+	if len(platform) > 0 {
+		buildctlArgs = append(buildctlArgs, "--opt=platform="+strings.Join(platform, ","))
+	}
+
+	buildArgsValue := options.BuildArgs
+	for _, ba := range strutil.DedupeStrSlice(buildArgsValue) {
+		arr := strings.Split(ba, "=")
+		if len(arr) == 1 && len(arr[0]) > 0 {
+			// Avoid masking default build arg value from Dockerfile if environment variable is not set
+			// https://github.com/moby/moby/issues/24101
+			val, ok := os.LookupEnv(arr[0])
+			if ok {
+				buildctlArgs = append(buildctlArgs, fmt.Sprintf("--opt=build-arg:%s=%s", ba, val))
+			} else {
+				logrus.Debugf("ignoring unset build arg %q", ba)
+			}
+		} else if len(arr) > 1 && len(arr[0]) > 0 {
+			buildctlArgs = append(buildctlArgs, "--opt=build-arg:"+ba)
+
+			// Support `--build-arg BUILDKIT_INLINE_CACHE=1` for compatibility with `docker buildx build`
+			// https://github.com/docker/buildx/blob/v0.6.3/docs/reference/buildx_build.md#-export-build-cache-to-an-external-cache-destination---cache-to
+			if strings.HasPrefix(ba, "BUILDKIT_INLINE_CACHE=") {
+				bic := strings.TrimPrefix(ba, "BUILDKIT_INLINE_CACHE=")
+				bicParsed, err := strconv.ParseBool(bic)
+				if err == nil {
+					if bicParsed {
+						buildctlArgs = append(buildctlArgs, "--export-cache=type=inline")
+					}
+				} else {
+					logrus.WithError(err).Warnf("invalid BUILDKIT_INLINE_CACHE: %q", bic)
+				}
+			}
+		} else {
+			return "", nil, false, "", nil, nil, fmt.Errorf("invalid build arg %q", ba)
+		}
+	}
+
+	labels := options.Label
+	labels = strutil.DedupeStrSlice(labels)
+	for _, l := range labels {
```

Comment:
We can avoid these redundant assignment code and use `options.Label` directly (same as below, e.g., `noCache := options.NoCache`, etc)

```golang
for _, l := range strutil.DedupeStrSlice(options.Label) {
...
```

(meanwhile, is `strutil.DedupeStrSlice` a typo? with an extra `Dedup(e)`)

Reply:
https://github.com/containerd/nerdctl/blob/main/cmd/nerdctl/build.go#L399-L406 

old code too.. Anyway, I will update it